Sol.java

```diff
@@ -0,0 +1,104 @@
+import java.util.*;
+
+public class Sol {
+    int[] in ;
+    HashMap < Integer, Integer > exp;
+    LinkedList < Move > moves_coll;
+    int zeropos;
+
+    public LinkedList<Move> move_cars(int[] in , int[] out) {
+        initialise( in , out);
+        solve();
+      
+        return moves_coll;
+    }
+
+    private void initialise(int[] in , int[] out) {
+        exp = new HashMap < Integer, Integer > ();
+        moves_coll = new LinkedList < Move > ();
+        this.in = in ;
+
+        for (int i = 0; i < out.length; i++) {
+            exp.put(out[i], i);
+        }
+
+        for (int j = 0; j < in .length; j++)
```

Comment:
Please :D no space before method invocation :D try to be more consistent with space usage e.g. line 26 pace after '(' is not necessary :)